app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt

```diff
@@ -209,62 +206,44 @@ class ReportingRepositoryImpl(
                 quarantineRepository.markMissingExposureKeysAsUploaded()
             } else {
                 // The regular flow of reporting the exposure keys.
-                if (infectionLevel == MessageType.InfectionLevel.Red) {
-                    val sentTemporaryExposureKeys =
-                        infectionMessengerRepository.getSentTemporaryExposureKeysByMessageType(
-                            MessageType.InfectionLevel.Yellow
-                        )
+                val thresholdTime = if (infectionLevel == MessageType.InfectionLevel.Red) {
```

Comment:
This method has gotten way to long IMHO, please split it into smaller methods that can be easily understood

Reply:
Will be done after rlease